ERCS/erc-xxxx.md

```diff
@@ -0,0 +1,201 @@
+
+## Motivation
+
+## Specification
+
+### Shared configuration
+
+All actions in Account Abstraction within the context of EIP-5792 must be done on a single chain and atomically.
+This means all requests to the `wallet_sendCalls` methods MUST be done:
+
+1. With the `atomicBatch` capability enabled
+2. With the `chainId` set to the same value in all calls
+
+### Static Paymaster Configuration Capability
+
+Note that use of Paymasters managed by a "paymaster web service" is described in [ERC-7677](./eip-7677).
+
+Identifier:
+
+`staticPaymasterConfiguration`
+
+Interface:
+
+```typescript
+type StaticPaymasterConfigurationCapabilityParams = Record<
+  `0x${string}`, // Chain ID
+  {
+    paymaster: string;
+    paymasterData: string;
+    paymasterValidationGasLimit: `0x${string}`;
+    paymasterPostOpGasLimit: `0x${string}`;
+  }
+>;
+```
+
+### On-chain Query Paymaster Configuration Capability
+
+Identifier:
+
+`onChainQueryPaymasterConfiguration`
+
+Interface:
+
+```typescript
+type OnChainQueryPaymasterConfigurationCapabilityParams = Record<
+  `0x${string}`, // Chain ID
+  {
+    target: `0x${string}`, // contract to query for Paymaster configuration
+    context: `0x${string}`, // hex-encoded byte array to pass to the configuration provider
+  }
+>
+```
+
+We then define the following Solidity interface:
+
+```solidity
+
+    struct PaymasterConfiguration {
+        address paymaster;
+        bytes paymasterData;
+        uint256 paymasterValidationGasLimit;
+        uint256 paymasterPostOpGasLimit;
+    }
+
+interface IPaymasterConfigurationResolver {
+    function getPaymasterConfiguration(bytes operation, bytes context) external;
+}
+
+```
+
+The wallet MUST perform an ABI-encoding of the entire `operation` object and make a view call
+to the `getPaymasterConfiguration` function of the `target` address.
+
+If the view call fails or returns an invalid data, the `wallet_sendCalls` method must fail and return with an error.
+
+### Validity Time Range Capability
+
+Identifier:
+
+`validityTimeRange`
+
+Interface:
+
+```typescript
+type ValidityTimeRangeCapabilityParams = Record<
+  `0x${string}`, // Chain ID
+  {
+    validAfter: `0x${string}`, // operation valid only after this timestamp, in seconds
+    validUntil: `0x${string}`  // operation valid only before this timestamp, in seconds
+  }
+>
+```
+
+The wallet must then verify the time range [`validAfter`..`validUntil`] is valid and present it to the
+user in a human-readable way for confirmation as part of the transaction information.
+
+### Multidimensional Nonce Capability
+
+Identifier:
+
+`multiDimensionalNonce`
+
+Interface:
+
+```typescript
+type MultiDimensionalNonceCapabilityParams = Record<
+  `0x${string}`, // Chain ID
+  {
+    nonceKey: `0x${string}`,
+    nonceSequence: `0x${string}`
+  }
+>
+```
+
+For Smart Contract Accounts that support multidimensional nonce values,
+the wallet must specify these parameters during the actual on-chain execution of the batch.
```

Comment:
The wallet MAY specify a `nonceKey`, (if it doesn't specify, then the default 0 is used)
nonceSequence is trickier: it requires an `eth_getNonce` API

Reply:
We probably need such an API. However, it can be hacked by calling the Nonce Manager directly, but that also gets very version - specific.